### DISCUSSION TITLE
Set tags on ipaddress over netboxshell

### DISCUSSION THREAD

**Discussion by julianstolp** [netbox-community/netbox#9531] [Q&A] [answered]
Hello, I'am trying to set a tag to an ipaddress over the netboxshell. But I don't get it. I checked all methods below inside the Netbox UI but none of them are working. 

> from ipam.models import *
> from extras.models import *
> ip_obj = IPAddress.objects.filter(address='10.10.10.10/24').first()
> ip_obj

10.10.10.10/24
> dir(ip_obj)

['DoesNotExist', 'Meta', 'MultipleObjectsReturned', '__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getstate__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__setstate__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_check_column_name_clashes', '_check_constraints', '_check_default_pk', '_check_field_name_clashes', '_check_fields', '_check_id_field', '_check_index_together', '_check_indexes', '_check_local_fields', '_check_long_column_names', '_check_m2m_through_same_relationship', '_check_managers', '_check_model', '_check_model_name_db_lookup_clashes', '_check_ordering', '_check_property_name_related_field_accessor_clashes', '_check_single_primary_key', '_check_swappable', '_check_unique_together', '_do_insert', '_do_update', '_get_FIELD_display', '_get_expr_references', '_get_next_or_previous_by_FIELD', '_get_next_or_previous_in_order', '_get_pk_val',
'_get_unique_checks', '_meta', '_perform_date_checks', '_perform_unique_checks', '_prepare_related_fields_for_save', '_save_parents', '_save_table', '_set_mask_length', '_set_pk_val', '_state', 'address', 'assigned_object', 'assigned_object_id', 'assigned_object_type', 'assigned_object_type_id', 'cf', 'check', 'clean', 'clean_fields', 'clone_fields', 'created', 'custom_field_data', 'date_error_message', 'delete', 'description', 'dns_name', 'family', 'fhrpgroup', 'from_db', 'full_clean', 'get_absolute_url', 'get_custom_fields', 'get_deferred_fields', 'get_duplicates', 'get_role_color', 'get_role_display', 'get_status_color', 'get_status_display', 'id', 'interface', 'journal_entries', 'last_updated', 'mask_length', 'nat_inside', 'nat_inside_id', 'nat_outside', 'objects', 'pk', 'prepare_database_save', 'refresh_from_db', 'role', 'save', 'save_base', 'serializable_value', 'services', 'snapshot', 'status', 'tagged_items', 'tags', 'tenant', 'tenant_id', 'to_objectchange', 'unique_error_message', 'validate_unique', 'vminterface', 'vrf', 'vrf_id']

>ip_obj.tags

ipam.IPAddress.None
> ip_obj.tags = [72]
> ip_obj.save()
> ip_obj.tags

[72]

> ip_obj.tags = 72
> ip_obj.save()
> ip_obj.tags

72

> tag_obj = Tag.objects.get(pk=72)
> tag_obj

Check Me!
> ip_obj.tags = tag_obj
> ip_obj.save()
> ip_obj.tags

<Tag: Check Me!>

I tried it with the pynetboxapi and it works.

ip_obj = nb.ipam.ip_addresses.get(address='1010.10.10/24')
ip_obj.tags = [72] 
ip_obj.save()

Do I miss something?

**Comment by jeremystretch** [comment, marked as answer]
Tag assignments are a many-to-many relationship, wherein a separate table is used to track each tag assignment to an object, so these work a bit differently from concrete model fields. To assign tags, use the `add()` or `set()` method on the manager:

```
ip_obj.tags.add(tag_obj)
```

You can find complete documentation of the available methods in the [django-taggit docs](https://django-taggit.readthedocs.io/en/latest/api.html). Hope that helps!

**Reply by julianstolp** [reply]
@jeremystretch Thank you, it helped a lot.
